lychee-lib/src/types/error.rs

```diff
@@ -27,19 +27,8 @@ pub enum ErrorKind {
     #[error("Network error (reqwest): {0}")]
     ReqwestError(#[from] reqwest::Error),
     /// Network error while using Github API (via hubcaps)
-    #[error("Network error (hubcaps) {}", match .0 {
-        Some(e) => match e {
-            hubcaps::Error::Fault { code, error } => format!(": {} [{}]", error.message.clone(), code),
-            hubcaps::Error::RateLimit { .. } => "Hit the rate limit".to_string(),
-            hubcaps::Error::Codec(e) => e.to_string(),
-            hubcaps::Error::Reqwest(e) => e.to_string(),
-            hubcaps::Error::Url(e) => e.to_string(),
-            hubcaps::Error::IO(e) => e.to_string(),
-            hubcaps::Error::JWT(e) => e.to_string(),
-        }
-        None => "".to_string(),
-    })]
-    GithubError(#[from] Option<hubcaps::Error>),
+    #[error("Network error (hubcaps) {}", .0.as_ref().map_or(String::new(), std::string::ToString::to_string))]
```

Comment:
this line should probably say `octocrab` or perhaps not even mention the client at all.

Reply:
Right. Let's just say it's a "GitHub Client" error, hiding the implementation details.